### ISSUE TITLE
Chapel does not yet support LLVM 15

### ISSUE THREAD

**Issue by ghbrown** [chapel-lang/chapel#21461]
### Summary of Problem

Since `clang` version 15 has become the main version distributed by the Arch Linux repositories, Chapel has been unable to build since it permits versions between 11 and 14.

I manage the Chapel packages for the Arch User Repository, and wanted to check in with the team on the best way to handle this. I optimistically tried to change [this line](https://github.com/chapel-lang/chapel/blob/8ca23c39b7f97e3f1a30d6a5e16242d2559b9ec8/util/chplenv/chpl_llvm.py#L18), but unfortunately I got an error (somewhat deep) in the build process.

The main options on my side are:
- constraining the package version for `clang` (I hear this is not so simple, but may be good for robustness of the package in the long run)
- personally downgrade my `clang` version and put temporary warnings on the package page (I really don't like this option)

Build commands:
```
./configure --prefix=/usr
make
```

I'll be investigating the dependency constraints, but if anybody has suggestions that would be great too.

### Configuration Information

- Output of `chpl --version`: `<currently erroring, was running 1.30 pre-release>`
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`: `clang version 15.0.7`

**Comment by ghbrown** [comment]
I'm almost positive it's the former, since the package builds nearly the default configuration, which I believe is the LLVM back-end these days. I think I've found a stop gap solution for now, but we'll see.

 Also, apologies if this was a duplicate, I didn't see any similar open issues.

**Comment by bradcray** [comment]
[oops: I did not mean to close this when adding my comment—I must have hit the wrong button!  I was going to propose we just convert this into the public-facing issue for LLVM 15 which is what made me realize I had]

No need to apologize!  I didn't see an open public issues either—we've got a bunch in our private repo's issues list and I hadn't realized that there wasn't a public-facing version until now.

**Comment by mppf** [comment]
@ghbrown - AFAIK what you will have to do for AUR is one of these three:

 1. Provide a package built with CHPL_LLVM=bundled (this will almost certainly significantly increase build times)
 2. Provide a package built with CHPL_LLVM=none (using the C backend) until this is sorted out (the drawback being this is not intended to be the default configuration)
 3. Wait to upgrade AUR until we can complete the work to upgrade to LLVM 15. It looks like that will not happen in 1.30 (coming up this spring) but I am hopeful we will complete it for 1.31 (in the fall).

**Comment by ghbrown** [comment]
Thank you both for the suggestions, I had forgotten about some of these.

For reference (I just learned these things myself):
- when new dependencies like `llvm` get a new version, all of the dependent Arch packages are rebuilt, and the user is expected to move to all of these new versions simultaneously with the new version of `llvm` (this means it's not possible just to downgrade the `llvm` and `clang` packages, since some very important packages will at least partially break)
- to help ease transitions like this, the Arch repositories contain versioned `llvm`s and `clang`s ([like this](https://archlinux.org/packages/extra/x86_64/clang14/)).

My first try with the `llvm14` group wasn't successful (unsure why), but I tried again and everything seems to be working now. I'll drop back to the bundled `llvm` if issues persist.
